### DISCUSSION TITLE
Integration tests for agent extensions

### DISCUSSION THREAD

**Discussion by daniloqueiroz** [open-telemetry/opentelemetry-java-instrumentation#5283] [Q&A] [answered]
Hello,
Is there any guide on how to test agent extensions together with the agent itself?

We have a couple of custom extensions, and when migrating the agent from `v1.7.2` to `v1.10.x`, we had an issue with one of the extensions, which caused the agent not to be initialized properly. 
So, although we've already addressed the root cause, we'd like to set up some integration tests to check whether or not our extensions work as expected with a given version of the OTel-Agent.

I've found [this](https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/4560), which seems somewhat related to this, but there are no updates on that issue either.

**Comment by iNikem** [comment, marked as answer]
Our own extension examples have an end-to-end integration test: https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/examples/extension/src/test/java/com/example/javaagent/smoketest. Does this help you?

**Reply by daniloqueiroz** [reply]
Yes, this is very helpful, thanks!
